tests/test_models.py

```diff
@@ -0,0 +1,32 @@
+from tests.factories import ExtraDataFactory
+
+
+def test_model_serialization(profile, snapshot):
+    profile.user.first_name = "First"
+    profile.user.save()
+    ExtraDataFactory(profile=profile)
+    serialized_profile = profile.serialize()
+
+    expected = [
+        {"key": "MEMO", "value": profile.memo},
+        {
+            "key": "USER",
+            "value": profile.user.first_name,
+        },
+    ]
+
+    expected_related = ["EXTRA_DATA"]
+
+    assert "key" in serialized_profile
+    assert "children" in serialized_profile
+    assert serialized_profile["key"] == "PROFILE"
+    assert len(serialized_profile["children"]) == len(expected) + len(expected_related)
+
+    for d in expected:
+        assert d in serialized_profile["children"]
+
+    # Check that related objects are included
+    for key in expected_related:
+        assert any(map(lambda x: x["key"] == key, serialized_profile["children"]))
```

Comment:
Isn't this bit basically duplicating the comparison on line 32 where you're using the snapshot? Or am I missing something crucial?

I would assume that the same test could be achieved by writing this as:

```
...
serialized_profile = profile.serialize()
snapshot.assert_match(serialized_profile)
```

Reply:
Ah, I actually had the test without the snapshot first, since a similar test existed on youth-membership. I then decided to add the `snapshot`, but it seems I forgot to refactor/remove the redundant code. :facepalm: